test/prefer-t-regex.js

```diff
@@ -62,6 +62,26 @@ ruleTester.run('prefer-t-regex', rule, {
 			code: header + 'const reg = /\\d+/;\ntest(t => t.true(reg.test(foo.bar())));',
 			output: header + 'const reg = /\\d+/;\ntest(t => t.regex(foo.bar(), reg));',
 			errors: errors('regex')
+		},
+		{
+			code: header + 'test(t => t.is(foo(), /\\d+/));',
```

Comment:
Can you also add version of these tests using `new RegExp()`?

Reply:
Do we want to expand the functionality for this? The original regex detection logic only matches regex literals, not expressions resolving to a regex. Resolving whether or not any given expression resolves to a regex pretty much requires eval, so we'd have to special-case `new RegExp()`.

Reply:
We don't need to resolve expression, but we should support an inline `new RegExp()` call.